scripts/sync-interface-data.js

```diff
@@ -0,0 +1,385 @@
+  }
+
+  const rows = await trx.insert({
+    origin_id: sourceData.id,
+    ...targetDataFactory(sourceData, ...args)
+  }).into(table).returning('id');
+
+  return rows[0];
+}
+
+/**
+ * Updates a row in the target database, based on transforming data from the source database.
+ *
+ * The provided source data must include an `id` property.
+ * The updated row in the target table will be the one with a matching `origin_id` column.
+ *
+ * @param {Transction} trx - Knex transaction.
+ * @param {string} table - The name of the table in which to update a row.
+ * @param {Function} targetDataFactory - Factory which, when called with this function's remaining arguments,
+ *                                       will produce the desired state of the row in the target table.
+ * @param {Object} sourceData - The data from the source database.
+ * @param {...*} args - Additional arguments to pass to the target data factory function.
+ */
+async function updateRecord(trx, table, targetDataFactory, sourceData, ...args) {
+  if (!sourceData.id) {
+    throw new Error(`Source data must have an ID: ${JSON.stringify(sourceData)}`);
+  }
+
+  await trx.update({
+    ...targetDataFactory(sourceData, ...args)
+  }).into(table).where('origin_id', sourceData.id);
+}
+
+/**
+ * Returns an action row for the target database (without identifiers), based on an action row from the source database.
+ *
+ * @param {Object} sourceData - The action in the source database.
+ * @param {Object} themeIdMap - Map with source theme IDs as keys and the corresponding target theme IDs as values.
+ * @returns {Object} The desired state of the action in the target database.
+ */
+function getActionTargetData(sourceData, themeIdMap) {
+  return {
+    ...pick(sourceData, 'code', 'title', 'created_at', 'updated_at'),
+    description: sourceData.short_description,
+    theme_id: themeIdMap[sourceData.theme_id]
+  };
+}
+
+/**
+ * Returns a theme row for the target database (without identifiers), based on a theme row from the source database.
+ *
+ * @param {Object} sourceData - The theme in the source database.
+ * @returns {Object} The desired state of the theme in the target database.
+ */
+function getThemeTargetData(sourceData) {
+  return {
+    ...pick(sourceData, 'code', 'title', 'description', 'created_at', 'updated_at'),
+    source: sourceData.source || null
+  };
+}
+
+/**
+ * Returns a representation of the specified PostgreSQL connection URL for logging.
+ *
+ * The userinfo component of the URL (username and password) is not included.
+ *
+ * @param {string} url - A full PostgreSQL connection URL.
+ * @returns {string} A simplified connection URL to display.
+ */
+function describeDbUrl(url) {
+  return `${url.hostname}:${url.port || '5432'}${url.pathname}`;
+}
+
+/**
+ * @typedef {Object} SyncSourceData
+ * @property {Object} actions - Action database rows.
+ * @property {Object} themes - Theme database rows.
+ */
```

Comment:
I don't understand this lonely comment (and the one below). Is this for documentation only, and if so, why didn't we use this notation anywhere else before ?

Reply:
It's a JSDoc feature that allows you to document a complex object with multiple properties as a return value, such as the return values of the `loadSourceData` and `loadTargetData` methods. I believe this was simply the first time I saw a need for it.